Added Husky, eslint, and prettier

TyGriffin115 opened the pull request:
Fixed the props validation errors. This branch should be merged last!!!

TyGriffin115 committed:
Added Husky, eslint, and prettier

powellsl99 commented on Ghost_in_the_Shell/client_folder_react_app/ethisim/.eslintcache:
```diff
@@ -0,0 +1 @@
+[{"C:\\Users\\tgrif\\Documents\\GitHub\\320-F20-Track-III\\Ghost_in_the_Shell\\client_folder_react_app\\ethisim\\src\\App.js":"1","C:\\Users\\tgrif\\Documents\\GitHub\\320-F20-Track-III\\Ghost_in_the_Shell\\client_folder_react_app\\ethisim\\src\\App.test.js":"2","C:\\Users\\tgrif\\Documents\\GitHub\\320-F20-Track-III\\Ghost_in_the_Shell\\client_folder_react_app\\ethisim\\src\\components\\AddNewSc ...
```
can you add this to the .gitignore and remove it from git

powellsl99 commented on Ghost_in_the_Shell/client_folder_react_app/ethisim/.prettierrc.json:
```diff
@@ -0,0 +1 @@
+{}
```
Should discuss some things to put in here. Here is a good starting point (definitely add or delete as you find useful!)
{
  "trailingComma": "es5",
  "tabWidth": 4,
  "semi": true,
  "singleQuote": true
}

powellsl99 commented on Ghost_in_the_Shell/client_folder_react_app/ethisim/10:
```diff
@@ -0,0 +1,8 @@
++ lint-staged@10.4.0
```
what is this file?

TyGriffin115 replied:
Nevermind, actually I am not sure. I took it out and everything seems to run fine so far.

TyGriffin115 committed:
Fixed props validation errors in data.js

Also fixed this error in scenarioCard.js but during runtime there's a bug where the name of the scenario does not pop up on each of the cards

TyGriffin115 committed:
Fixed runtime bug missing scenarioName

eneadodi commented:
Am i suppose to be reviewing that husky prettier and Eslint are working correctly? If they have been successfully installed, I can simply accept the pull request. 
I think Sean would be a better option here.

powellsl99 reviewed (changes requested):
Haven't tried running it yet, but I have a few comments. Also there are a lot of merge conflicts. Let me know when this is done and I can run it and re-review!

TyGriffin115 committed:
Test

Removed the weird 10 file
Added .eslintcache to gitignore
Added recomended items to prettierrc.json file

TyGriffin115 committed:
Merge branch 'master' into 101-Implement-Husy-Prettier-ESLINT-2

TyGriffin115 committed:
Fixed new Errors from merging master into this branch

Fixed more prop-validation errors
Fixed "missing key for prop elem error" errors

TyGriffin115 commented:
I thought it would be easier to create a new branch from the updated master and install prettier-husky-eslint, then to deal with the merge conflicts.
